File: ImmoMange-ui/src/app/services/models/property-response-dto.ts
/* tslint:disable */
/* eslint-disable */
import { PropertyEquipmentDto } from '../models/property-equipment-dto';
import { PropertyImageDto } from '../models/property-image-dto';
export interface PropertyResponseDto {
  address?: string;
  cinTenants?: Array<string>;
  description?: string;
  id?: number;
  managerEmail?: string;
  propertyEquipmentDto?: Array<PropertyEquipmentDto>;
  propertyNames?: Array<string>;
  rentPrice?: number;
  status?: 'OCCUPIED' | 'AVAILABLE' | 'UNDER_MAINTENANCE';
}

File: src/main/java/org/asm/immomanage/dto/propertyDto/PropertyResponseDto.java
package org.asm.immomanage.dto.propertyDto;

import lombok.Builder;
import org.asm.immomanage.models.*;
import org.asm.immomanage.utils.Status;
import java.util.Set;

@Builder
public record PropertyResponseDto(
        long id,
        double rentPrice,
        String description,
        String address,
        Status status,
        Set<String> propertyEquipments,
        Set<String> propertyNames,
        Set<Long> tenantsIDS,
        Long managerID
        ){}




